### PR TITLE
 DOP-4461: CI rstspec.toml 

### PR DESCRIPTION
**Pull request by i80and**
### Ticket

 DOP-4461

### Notes

Failed push tested in https://github.com/mongodb/snooty-parser/actions/runs/8625012846/job/23640921597?pr=581

### README updates

- - [ ] This PR introduces changes that should be reflected in the README.md and/or HACKING.md, and I have made those updates.
- - [x] This PR does not introduce changes that should be reflected in the README.md and/or HACKING.md

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -3,59 +3,27 @@ name: CI
...
-          token: ${{ secrets.GITHUB_TOKEN }}
-          draft: true
-          prerelease: true
+    - name: Fetch parser
+      run: |
+        mkdir -p downloaded_parser
+        curl -L "$(python3 ./tools/get_latest_parser_url.py)" -o downloaded_parser/snooty.zip
+        cd downloaded_parser && unzip snooty.zip
+    - name: Test parser
+      run: mkdir empty && ./downloaded_parser/snooty/snooty build --rstspec=snooty/rstspec.toml empty/
```
Comment (rayangler) [inline .github/workflows/test.yml:29]: (Just curious; might be out of scope) Would we ever want to run the parser tests with the rstspec file in this branch to flush out any inconsistencies between the rstspecs? Kind of an extreme example, but if `literalinclude` is removed from the latest rstspec, parsing this empty directory passes, but the rstspec is technically flawed.

Comment (i80and) [reply .github/workflows/test.yml:29]: Not a bad idea, actually!

Comment (i80and) [reply .github/workflows/test.yml:29]: I don't think that needs to block this, though. It can be a follow-up ticket if we decide to do that.

Comment (rayangler) [reply .github/workflows/test.yml:29]: Oh, yeah, for sure! Sorry, I should've approved before

**Review by rayangler — APPROVED** [review]

**Pull request merged**